section_faiss_vector_index.go

```diff
@@ -153,18 +154,16 @@ LOOP:
 				}
 
 				// remap the docIDs from the old segment to the new document nos.
-				// provided.
+				// provided. furthermore, this function also drops the invalid doc nums
+				// of that segment in the resulting bitmap
 				bitMap = remapDocIDs(bitMap, newDocNumsIn[segI])
 				if vecToDocID[vecID] == nil {
-					// if there are some tombstone entries in the docIDs, marked
-					// in the drops[ith-segment] bitmap, don't include them in the
-					// final bitmap.
-					if drops[segI] != nil && !drops[segI].IsEmpty() {
-						vecToDocID[vecID] = roaring.AndNot(bitMap, drops[segI])
-					} else {
+					if bitMap.GetCardinality() > 0 {
 						vecToDocID[vecID] = bitMap
+						indexes[len(indexes)-1].vecIds = append(indexes[len(indexes)-1].vecIds, vecID)
```

Comment:
I'd advise some commentary here on why the particular index of the `indexes` array.